mysql-test/suite/binlog_encryption/my.cnf

```diff
@@ -1,5 +1,4 @@
-!include include/default_mysqld.cnf
-!include include/default_client.cnf
+!include include/default_my.cnf
```

Comment:
Appears that the vast majority of the changes in this PR are due to replacing `default_mysqld.cnf` and/or `default_client.cnf` with `default_my.cnf`… rather than with the stated purpose of addressing https://jira.mariadb.org/browse/MDEV-30587.

The actual changes for MDEV-30587 are addressed mainly in the changes to `userstat.{test,result}` and `wait_until_connected_again.inc`… along with a few other files (`ha.result` and `rpl*.test`) which are _very_ easy to overlook amidst all the include file replacements.

At least to me, it's not apparent from this PR…

1. If [`default_my.cnf`](https://github.com/MariaDB/server/blob/11.1/mysql-test/include/default_my.cnf) is in fact "equivalent" to the combination of `default_mysqld.cnf` and `default_client.cnf`, or if it differs from them in some way.
2. Why this change is being made here.
3. Whether it's related in some way to fixing the test result mismatches caused by MTR connecting with TLS. (I think the answer is **no**, but I'm definitely not 100% certain.)

Given all that, shouldn't these changes be split out into a separate PR, or a separate commit, with a separate description for what's being changed and why?

Reply:
@vuvova picked this up in b0fc90ac8a64486ede34999c7e3a2dea9bc5fc46